packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -286,9 +286,8 @@ export abstract class TextFieldBase extends FormElement {
       this.createFoundation();
     }
 
-    if (changedProperties.has('value') &&
-        this.value !== undefined) {
-      this.mdcFoundation.setValue(this.value);
+    if (changedProperties.has('value')) {
+      this.mdcFoundation.setValue(this.value === undefined ? '' : this.value);
```

Comment:
this treats undefined values as empty strings and avoids 
![image](https://user-images.githubusercontent.com/1168053/79444393-678bc500-7fdb-11ea-9b04-b7abe880cedc.png)

Reply:
This is actually the exact type of behavior I want us to avoid doing. This adds logic to support setting a property to an invalid type. According to typescript, `this.value` should never be `undefined` because it must be a strict String type.

We don't have any extra logic to handle `this.value = new Date()`, even though it could be a valid use case for `type="date"` inputs. It'd be an ever-growing maintenance cost to continually add logic for all the different types a property could be set to. If there's a valid use case for the property to be undefined, it should be typed as `string | undefined`.

If we're wanting to achieve native `<input>` parity, which uses a strict `string` for it's value (it can't be undefined), then we should be coercing all values to a string. Setting `textfield.value = undefined` would be the same as `textfield.value = String(undefined)`.